python/pyarrow/_dataset.pyx

```diff
@@ -3337,7 +3337,7 @@ cdef class Scanner(_Weakrefable):
         ----------
         dataset : Dataset
             Dataset to scan.
-        columns : list of str, default None
+        columns : list of str or dict, default None
```

Comment:
It seems that this may be interpreted as `list of (str or dict)` not `(list of str) or (dict)`.
Can we use clearer wording here?

Reply:
We could use `dict or list of str`. It doesn't seem like there is precedence in the Arrow docs to use `list[str] or dict`, which could also be nice.

Reply:
I also saw `list of str or dict, default None` being used in the document. I like the `list[str]`
```suggestion
        columns : list[str] or dict, default None
```